packages/elements/src/photon-med-search/components/MedicationFilterDropdown.tsx

```diff
@@ -90,25 +90,39 @@ export const MedicationFilterDropdown = (props: MedicationFormDropdownProps) =>
     }
   });
 
+  createEffect(() => {
+    if (selected()?.id) {
+      setFilterId(selected()!.id);
+      props.setMedicationId(selected()!.id);
```

Comment:
This feels like it could end up with two different event loops. I think it might be a good idea to use [`batch`](https://www.solidjs.com/tutorial/reactivity_batch)